datafusion/optimizer/src/eliminate_filter.rs

```diff
@@ -62,6 +62,14 @@ impl OptimizerRule for EliminateFilter {
                     }))),
                 }
             }
+            LogicalPlan::Filter(Filter {
+                predicate: Expr::Literal(ScalarValue::Boolean(None)),
+                input,
+                ..
+            }) => Ok(Some(LogicalPlan::EmptyRelation(EmptyRelation {
```

Comment:
How about merging it into the previous `false` match arm like this 🤔?
```rust
fn try_optimize(
        &self,
        plan: &LogicalPlan,
        _config: &dyn OptimizerConfig,
    ) -> Result<Option<LogicalPlan>> {
        match plan {
            LogicalPlan::Filter(Filter {
                predicate: Expr::Literal(ScalarValue::Boolean(v)),
                input,
                ..
            }) => {
                match *v {
                    // input also can be filter, apply again
                    Some(true) => Ok(Some(
                        self.try_optimize(input, _config)?
                            .unwrap_or_else(|| input.as_ref().clone()),
                    )),
                    Some(false) | None => {
                        Ok(Some(LogicalPlan::EmptyRelation(EmptyRelation {
                            produce_one_row: false,
                            schema: input.schema().clone(),
                        })))
                    }
                }
            }
            _ => Ok(None),
        }
    }
```

Reply:
Updated